google-cloud-pubsub/test/google/cloud/pubsub/topic/publish_async_test.rb

```diff
@@ -79,6 +79,44 @@
     _(callback_called).must_equal true
   end
 
+  it "publishes multiple messages with callbacks" do
+    messages = [
+      Google::Cloud::PubSub::V1::PubsubMessage.new(data: "async-message 0".encode(Encoding::ASCII_8BIT), message_id: "msg0"),
+      Google::Cloud::PubSub::V1::PubsubMessage.new(data: "async-message 1".encode(Encoding::ASCII_8BIT), message_id: "msg1")
+    ]
+    callbacks_called = 0
+
+    topic.service.mocked_publisher = AsyncPublisherStub.new
+
+    _(topic.async_publisher).must_be :nil?
+
+    topic.publish_async "async-message 0" do |result|
+      assert_equal "msg0", result.msg_id
+      callbacks_called += 1
```

Comment:
Should the common counter be protected by a lock? If the two messages are published at the same time, the threads incrementing the counter could interfere with each other, and the (false) failure could be difficult to reproduce.

Reply:
Fixed, thanks!